steps/s3/session.go

```diff
@@ -0,0 +1,179 @@
+	}
+	return nil
+}
+
+// uploadS3FileWithContent creates a new file in S3 with the content specified.
+func (s *Session) uploadS3FileWithContent(ctx context.Context, bucket, key, message string) error {
+	uploader := s3manager.NewUploader(s.Client)
+	_, err := uploader.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+		Body:   strings.NewReader(message),
+	})
+	if err != nil {
+		return fmt.Errorf("unable to upload %q to %q, %v", key, bucket, err)
+	}
+
+	s.CreatedDocuments = append(s.CreatedDocuments, &CreatedDocument{bucket: bucket, key: key})
+	GetLogger().Log("upload", bucket, key)
+	return nil
+}
+
+// createBucket creates a new bucket.
+func (s *Session) createBucket(ctx context.Context, bucket string) error {
+	logrus.Debugf("Creating a new bucket: %s", bucket)
+	cparams := &s3.CreateBucketInput{
+		Bucket: aws.String(bucket),
+	}
+
+	s3Client := s3.New(s.Client)
+	_, err := s3Client.CreateBucket(cparams)
+	if err != nil {
+		logrus.Errorf("Error creating a new bucket: %s, err: %v", bucket, err)
+	}
+	return nil
+}
+
+// validateS3File checks the existence of a file in S3.
+func (s *Session) validateS3File(ctx context.Context, bucket, key string) error {
+	logrus.Debugf("Validating s3 file exists bucket '%s' key '%s'", bucket, key)
+	s3svc := s3.New(s.Client)
+	exists, err := s3KeyExists(s3svc, bucket, key)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("failed validating s3 file exits: no file exists in bucket '%s' with name '%s' ", bucket, key)
+	}
+	return nil
+}
+
+// validateS3FileWithContent checks the existence of a file in S3 with the content specified.
+func (s *Session) validateS3FileWithContent(ctx context.Context, bucket, key, message string) error {
+	expected := strings.TrimSpace(message)
+	logrus.Debugf("Validating s3 file content bucket '%s' key '%s' content:\n%s", bucket, key, expected)
+	downloader := s3manager.NewDownloader(s.Client)
+	buf := aws.NewWriteAtBuffer([]byte{})
+	_, err := downloader.Download(buf, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		return fmt.Errorf("unable to upload %q to %q, %v", key, bucket, err)
+	}
+	actual := strings.TrimSpace(string(buf.Bytes()))
+	if expected != actual {
+		return fmt.Errorf("failed validating s3 bucket '%s' file '%s' content: expected:\n%s\n\nactual:\n%s", bucket, key, expected, actual)
+	}
+	return nil
+}
+
+// deleteS3File deletes the file in S3.
+func (s *Session) deleteS3File(ctx context.Context, bucket, key string) error {
+	logrus.Debugf("Delete s3 file in bucket '%s' key '%s'\n", bucket, key)
+	s3svc := s3.New(s.Client)
+	input := &s3.DeleteObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	}
+	_, err := s3svc.DeleteObject(input)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// s3KeyExists checks the existence of a key in a S3 bucket.
+func s3KeyExists(s3svc *s3.S3, bucket string, key string) (bool, error) {
```

Comment:
May change function? So it is a S3Session method and can be externally used

Reply:
Sure, you're right